### PR TITLE
Add Kyma environment broker component

### PR DESCRIPTION
**Pull request by polskikiel**
**Description**

Changes proposed in this pull request:

- Add Kyma environment broker component with chart

**Related issue(s)**

### CODE REVIEW

File: components/kyma-environment-broker/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Kyma Environment Service Broker
+
+## Overview
+
+The Kyma Environment Service Broker provide a way to run Kyma as a service on different providers like: AWS, GCP or Azure.
+
+The broker provides a Service Class which installs Kyma with 3 separate Service Plans for each provider.
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/README.md]: in the terms of OSB API, broker provides serivces (not service classes)

File: chart/compass/charts/kyma-environment-broker/values.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+replicaCount: 1
+
+image:
+  repository: eu.gcr.io/kyma-project/incubator/kyma-environment-broker
```
Comment (crabtree) [inline chart/compass/charts/kyma-environment-broker/values.yaml]: Why don't you just put those values regarding the image in the main values.yaml file?

File: components/kyma-environment-broker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+FROM alpine:latest as certs
+RUN apk --update add ca-certificates
+
+# Final image
+FROM scratch
+
+COPY --from=certs /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
+COPY --from=build /bin/kyma-env-broker /bin/kyma-env-broker
+
+ENV APP_PORT 8080
```
Comment (crabtree) [inline components/kyma-environment-broker/Dockerfile]: When you specify the env variable with a port that can be overridden during the run phase, why you then use the `EXPOSE` keyword that always will show 8080/tcp?

File: components/kyma-environment-broker/README.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# Kyma Environment Service Broker
+
+## Overview
+
+The Kyma Environment Service Broker provide a way to run Kyma as a service on different providers like: AWS, GCP or Azure.
+
+The broker provides a services which installs Kyma with a separate plan for each provider.
+
+It uses the Compass Provisioner API to install Kyma on the given cluster.
+
```
Comment (crabtree) [inline components/kyma-environment-broker/README.md]: Maybe you could write something more about how to run the broker itself? Some words about the parameters.

File: components/kyma-environment-broker/docs/README.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# Docs
+
+## Overview
+
+This folder contains documents that relate to the project.
+
+> In this document, provide the list of all documents that this folder contains, together with links to those documents and short information on what they describe.
+
+- `dictionary.md` contains a definitions of the specific
```
Comment (crabtree) [inline components/kyma-environment-broker/docs/README.md]: This sounds like there is missing some endings

File: components/kyma-environment-broker/docs/dictionary.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+## CIS and ERS
+
+![](assets/sm.png)
+
+Commercial Infrastructure Services (CIS) is a fundamental part of the SAP Cloud Platform Foundation, serving both commercials, technical and operational goals (read [this](https://wiki.wdf.sap.corp/wiki/display/CPC15N/%5BINTRO%5D+CIS@CF+Overview) for more info). One of sub-domain is **CIS Provisioning and registries (XRS, ERS, XPR, Tenant LCM)** which contains Environment Registry Service (ERS).	
+
+The ERS manages the lifecycle of environment instances like CF organizations, K8S clusters. During provisioning context is passed to the provisioned environment and mapping between environment instance and the commercial context is kept (binding to subaccount). See here more information about the environment manager  https://wiki.wdf.sap.corp/wiki/display/CPC15N/CIS+Environment+Registry+Service
+
+
+More about the ERS: https://wiki.wdf.sap.corp/wiki/pages/viewpage.action?pageId=1917617707
```
Comment (crabtree) [inline components/kyma-environment-broker/docs/dictionary.md]: Yet another internal link

File: components/kyma-environment-broker/docs/dictionary.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+Link: https://help.sap.com/viewer/65de2977205c403bbc107264b8eccf4b/Cloud/en-US/51ec15a8979e497fbcaadf80da9b63ba.html
+
+
+## CIS and ERS
+
+![](assets/sm.png)
+
+Commercial Infrastructure Services (CIS) is a fundamental part of the SAP Cloud Platform Foundation, serving both commercials, technical and operational goals (read [this](https://wiki.wdf.sap.corp/wiki/display/CPC15N/%5BINTRO%5D+CIS@CF+Overview) for more info). One of sub-domain is **CIS Provisioning and registries (XRS, ERS, XPR, Tenant LCM)** which contains Environment Registry Service (ERS).	
+
+The ERS manages the lifecycle of environment instances like CF organizations, K8S clusters. During provisioning context is passed to the provisioned environment and mapping between environment instance and the commercial context is kept (binding to subaccount). See here more information about the environment manager  https://wiki.wdf.sap.corp/wiki/display/CPC15N/CIS+Environment+Registry+Service
```
Comment (crabtree) [inline components/kyma-environment-broker/docs/dictionary.md]: Looks like another internal link.

File: components/kyma-environment-broker/docs/dictionary.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+Link: https://help.sap.com/viewer/65de2977205c403bbc107264b8eccf4b/Cloud/en-US/51ec15a8979e497fbcaadf80da9b63ba.html
+
+
+## CIS and ERS
+
+![](assets/sm.png)
+
+Commercial Infrastructure Services (CIS) is a fundamental part of the SAP Cloud Platform Foundation, serving both commercials, technical and operational goals (read [this](https://wiki.wdf.sap.corp/wiki/display/CPC15N/%5BINTRO%5D+CIS@CF+Overview) for more info). One of sub-domain is **CIS Provisioning and registries (XRS, ERS, XPR, Tenant LCM)** which contains Environment Registry Service (ERS).
```
Comment (crabtree) [inline components/kyma-environment-broker/docs/dictionary.md]: Looks like internal link in the text.

File: components/kyma-environment-broker/hack/scripts/README.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
+## Create Network and CloudNAT
```
Comment (crabtree) [inline components/kyma-environment-broker/hack/scripts/README.md]: Not sure this file should be here. For example, you use in the resources the domain name `*.cis.gophers.kyma.pro`. Maybe you can move this domain as an environment variable?

File: components/kyma-environment-broker/hack/scripts/vars.sh [hunk 10]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
```
Comment (crabtree) [inline components/kyma-environment-broker/hack/scripts/vars.sh]: Really want to commit this file?

In general, you could move your provisioning scripts to the installation directory of the whole compass.

**Comment by polskikiel** [comment]
@crabtree I removed the whole `/hack` folder as we don't need that anymore. I've also removed all of the internal content and applied your comments.

File: components/kyma-environment-broker/README.md [hunk 11]
```diff
@@ -0,0 +1,25 @@
+# Kyma Environment Service Broker
+
+## Overview
+
+The Kyma Environment Service Broker provide a way to run Kyma as a service on different providers like: AWS, GCP or Azure.
```
Comment (klaudiagrz) [inline components/kyma-environment-broker/README.md]:
```suggestion
Kyma Environment Broker provides a way to run Kyma as a Runtime on clusters provided by different cloud providers, such as AWS, GCP or Azure.
```

File: components/kyma-environment-broker/README.md [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+
+The broker provides a services which installs Kyma with a separate plan for each provider.
```
Comment (klaudiagrz) [inline components/kyma-environment-broker/README.md]:
```suggestion
The broker provides services that install Kyma with a separate plan for each provider.
```

File: components/kyma-environment-broker/README.md [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+
```
Comment (klaudiagrz) [inline components/kyma-environment-broker/README.md]: Please remove the gaps between sentences.

File: components/kyma-environment-broker/README.md [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+It uses the Compass Provisioner API to install Kyma on the given cluster.
```
Comment (klaudiagrz) [inline components/kyma-environment-broker/README.md]:
```suggestion
It uses the Compass Provisioner API to install Kyma on the given cluster. For more information, go to the [docs](../../docs/kyma-environment-broker/) directory.
```

File: components/kyma-environment-broker/README.md [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+
+## Configuration
```
Comment (klaudiagrz) [inline components/kyma-environment-broker/README.md]:
```suggestion
## Development
```

File: components/kyma-environment-broker/README.md [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+
+Use the following environment variables to configure the `Kyma Environment Service Broker`:
```
Comment (klaudiagrz) [inline components/kyma-environment-broker/README.md]:
```suggestion
Use the following environment variables to configure the Kyma Environment Service Broker:
```

File: components/kyma-environment-broker/README.md [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+
+| Name | Required | Default | Description |
+|-----|:---------:|--------|------------|
+| **APP_PORT** | No | `8080` | The port on which the HTTP server listens. |
+| **APP_PROVISIONING_URL** | No |  | Specifies an URL to the provisioner API. |
+| **APP_PROVISIONING_SECRET_NAME** | No | | Specifies the name of the Secret which holds credentials to the provisioner API. |
```
Comment (klaudiagrz) [inline components/kyma-environment-broker/README.md]:
```suggestion
| **APP_PROVISIONING_SECRET_NAME** | No | | Specifies the name of the Secret which holds credentials to the Provisioner API. |
```

File: components/kyma-environment-broker/README.md [hunk 18]
```diff
@@ -0,0 +1,25 @@
...
+| **APP_PROVISIONING_GARDENER_PROJECT_NAME** | No | `true` | Defines the used gardener project name. |
```
Comment (klaudiagrz) [inline components/kyma-environment-broker/README.md]:
```suggestion
| **APP_PROVISIONING_GARDENER_PROJECT_NAME** | No | `true` | Defines the used Gardener project name. |
```

File: components/kyma-environment-broker/README.md [hunk 19]
```diff
@@ -0,0 +1,25 @@
...
+| **APP_PROVISIONING_GCP_SECRET_NAME** | No | | Defines the name of the Secret which holds credentials to GCP. |
+| **APP_PROVISIONING_AWS_SECRET_NAME** | No | | Defines the name of the Secret which holds credentials to AWS. |
+| **APP_PROVISIONING_AZURE_SECRET_NAME** | No | | Defines the name of the Secret which holds credentials to AZURE. |
```
Comment (klaudiagrz) [inline components/kyma-environment-broker/README.md]:
```suggestion
| **APP_PROVISIONING_AZURE_SECRET_NAME** | No | | Defines the name of the Secret which holds credentials to Azure. |
```

File: docs/kyma-environment-broker/README.md [hunk 20]
```diff
@@ -0,0 +1,9 @@
+# Docs
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/README.md]:
```suggestion
# Kyma Environment Broker
```

File: components/kyma-environment-broker/README.md [hunk 21]
```diff
@@ -0,0 +1,25 @@
+# Kyma Environment Service Broker
```
Comment (klaudiagrz) [inline components/kyma-environment-broker/README.md]:
```suggestion
# Kyma Environment Broker
```

File: docs/kyma-environment-broker/README.md [hunk 22]
```diff
@@ -0,0 +1,9 @@
...
+
+## Overview
+
+This folder contains documents that relate to the project.
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/README.md]:
```suggestion
Kyma Environment Broker provides a way to run Kyma as a Runtime on clusters provided by different cloud providers, such as AWS, GCP or Azure. The broker provides services that install Kyma with a separate plan for each provider. It uses the Compass Provisioner API to install Kyma on the given cluster.
```

File: docs/kyma-environment-broker/README.md [hunk 23]
```diff
@@ -0,0 +1,9 @@
...
+
+> In this document, provide the list of all documents that this folder contains, together with links to those documents and short information on what they describe.
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/README.md]:
```suggestion

```

File: docs/kyma-environment-broker/README.md [hunk 24]
```diff
@@ -0,0 +1,9 @@
...
+
+- `dictionary.md` provides a definitions for a specific applications and/or platforms which interacts with the Kyma Environment Broker.
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/README.md]:
```suggestion

```

File: docs/kyma-environment-broker/dictionary.md [hunk 25]
```diff
@@ -0,0 +1,25 @@
+# Dictionary
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/dictionary.md]: I'm not sure whether this document should be put on the open-source GitHub as it contains many SAP-related words.

File: components/kyma-environment-broker/README.md [hunk 26]
```diff
@@ -0,0 +1,23 @@
+# Kyma Environment Broker
+
+## Overview
+
+Kyma Environment Broker provides a way to run Kyma as a Runtime on clusters provided by different cloud providers, such as AWS, GCP or Azure.
+The broker provides services that install Kyma with a separate plan for each provider.
+It uses the Compass Provisioner API to install Kyma on the given cluster. For more information, go to the [docs](../../docs/kyma-environment-broker/) directory.
```
Comment (klaudiagrz) [inline components/kyma-environment-broker/README.md]:
```suggestion
It uses the Compass Provisioner API to install Kyma on the given cluster.
```

File: components/kyma-environment-broker/README.md [hunk 27]
```diff
@@ -0,0 +1,23 @@
...
+
+## Development
+
+Use the following environment variables to configure the Kyma Environment Service Broker:
```
Comment (klaudiagrz) [inline components/kyma-environment-broker/README.md]:
```suggestion
Use the following environment variables to configure the Kyma Environment Broker:
```

**Pull request merged**